tests_typing/test_params.mypy-testing

```diff
@@ -0,0 +1,44 @@
+from typing import Any, Dict, Optional, Type, TypeVar, cast
+
+import pytest
+from pydantic import BaseModel
+from scrapy import Spider
+from scrapy.utils.test import get_crawler
+
+from scrapy_spider_metadata import Args
+from scrapy_spider_metadata._params import ParamSpecT
+
+
+SpiderT = TypeVar("SpiderT", bound=Spider)
+
+
+def get_spider(
+    spidercls: Type[SpiderT],
+    settings: Optional[Dict[str, Any]] = None,
+    kwargs=Optional[Dict[str, Any]]
+) -> SpiderT:
+    crawler = get_crawler(spidercls, settings or {})
+    spider = crawler._create_spider(spidercls.name, **(kwargs or {}))
+    return cast(SpiderT, spider)
+
+
+@pytest.mark.mypy_testing
+def test_paramspect():
+    class Params(BaseModel):
+        foo: int
+
+    class ParamSpider(Args[Params], Spider):
+        name = "params"
+
+    spider = get_spider(ParamSpider, kwargs={"foo": "1"})
+    reveal_type(spider)  # R: __main__.ParamSpider@30
+    reveal_type(spider.args)  # R: __main__.Params@27
```

Comment:
these @27 look extremely fragile; would it make sense to move Params and ParamSpider definitions outside the function body?